### PR TITLE
feat: Implement Component-Based Strategy Architecture Foundation

### PR DESCRIPTION
**Pull request by alexflorisca**
# 🏗️ Component-Based Strategy Architecture Foundation

This PR implements **Task 1** of the strategy system redesign, establishing the foundational component-based architecture that will replace the current monolithic strategy approach.

## 📋 Overview

This implementation introduces a modular, testable, and extensible strategy system with clear separation of concerns. The new architecture enables:

- **Composable Components**: Mix and match signal generators, risk managers, and position sizers
- **Regime Awareness**: All components adapt behavior based on market conditions  
- **Version Management**: A/B testing and rollback capabilities for strategy evolution
- **Comprehensive Testing**: 182 unit tests ensuring reliability and correctness

#### 1. SignalGenerator (`src/strategies/components/signal_generator.py`)
- **Abstract base class** with `generate_signal()` and `get_confidence()` methods
- **Signal dataclass** with direction, strength, confidence, and metadata validation
- **SignalDirection enum** (BUY, SELL, HOLD)
- **Test implementations**: `HoldSignalGenerator`, `RandomSignalGenerator`

#### 2. RiskManager (`src/strategies/components/risk_manager.py`)  
- **Abstract base class** with `calculate_position_size()`, `should_exit()`, `get_stop_loss()` methods
- **Position and MarketData dataclasses** with comprehensive validation
- **FixedRiskManager implementation** with regime-aware adjustments
- **Risk parameter validation** and bounds checking

#### 3. PositionSizer (`src/strategies/components/position_sizer.py`)
- **Abstract base class** with `calculate_size()` method
- **Multiple implementations**:
  - `FixedFractionSizer`: Simple percentage-based sizing
  - `ConfidenceWeightedSizer`: Size based on signal confidence
  - `KellySizer`: Kelly Criterion optimal sizing with trade history tracking
- **Utility functions** for risk-based position calculations

#### 4. RegimeContext (`src/strategies/components/regime_context.py`)
- **RegimeContext dataclass** with trend, volatility, confidence, duration, and strength
- **EnhancedRegimeDetector** with component-specific adaptations
- **RegimeTransition tracking** for regime change analysis
- **Regime history management** and stability detection

#### 5. StrategyManager (`src/strategies/components/strategy_manager.py`)
- **Component orchestration** with versioning capabilities
- **StrategyVersion and StrategyExecution** data models
- **A/B testing support** with performance comparison
- **Rollback capabilities** and execution history tracking

## 🧪 Comprehensive Test Suite

Added **182 unit tests** across 5 test files with 100% pass rate:

- `tests/strategies/components/test_signal_generator.py` (26 tests)
- `tests/strategies/components/test_risk_manager.py` (40 tests)  
- `tests/strategies/components/test_position_sizer.py` (50 tests)
- `tests/strategies/components/test_regime_context.py` (39 tests)
- `tests/strategies/components/test_strategy_manager.py` (27 tests)

### Test Coverage Includes:
- ✅ **Validation testing** for all data models and edge cases
- ✅ **Error handling** and boundary condition testing  
- ✅ **Mock implementations** for testing abstract interfaces
- ✅ **Integration testing** between components
- ✅ **Regime-aware behavior** testing

### Architecture Patterns
- **Dependency Injection**: Components are injected into StrategyManager
- **Strategy Pattern**: Multiple implementations of each component type
- **Observer Pattern**: Regime context passed between components
- **Version Control**: Full versioning system for strategy evolution

### Key Features
- **Input Validation**: Comprehensive validation throughout all components
- **Error Handling**: Graceful degradation and safe defaults
- **Regime Awareness**: All components adapt to market conditions
- **Performance Tracking**: Execution metrics and component attribution
- **Extensibility**: Easy to add new component implementations

## 📊 Requirements Satisfied

- ✅ **Requirements 1.1, 1.2, 1.3, 1.4**: Component interfaces implemented
- ✅ **Requirements 2.1, 2.2**: Signal generation and risk management
- ✅ **Requirements 6.1, 6.3, 6.4**: Component validation and testing
- ✅ **Requirements 2.4, 3.1, 3.2**: Regime detection and adaptation
- ✅ **Requirements 8.1, 8.2**: Comprehensive unit testing

## 🚀 Next Steps

This foundation enables the following upcoming tasks:
1. **Legacy Adapter System** (Task 2) - Backward compatibility
2. **Signal Generator Extraction** (Task 3) - Convert existing strategies  
3. **Risk Manager Implementation** (Task 4) - Advanced risk management
4. **Position Sizer Enhancement** (Task 5) - Sophisticated sizing algorithms
5. **Strategy Composition** (Task 6) - Complete strategy assembly

### New Components
- `src/strategies/components/__init__.py`
- `src/strategies/components/signal_generator.py`
- `src/strategies/components/risk_manager.py` 
- `src/strategies/components/position_sizer.py`
- `src/strategies/components/regime_context.py`
- `src/strategies/components/strategy_manager.py`

### New Tests
- `tests/strategies/components/__init__.py`
- `tests/strategies/components/test_signal_generator.py`
- `tests/strategies/components/test_risk_manager.py`
- `tests/strategies/components/test_position_sizer.py`
- `tests/strategies/components/test_regime_context.py`
- `tests/strategies/components/test_strategy_manager.py`

### Updated Documentation
- `.kiro/specs/strategy-system-redesign/tasks.md` (task completion tracking)

## ✅ Testing

```bash
# Run all component tests
python -m pytest tests/strategies/components/ -v

# Results: 182 passed, 31 warnings in 0.67s
```

## 🎉 Impact

This PR establishes the foundation for a **modern, maintainable, and extensible** strategy system that will:

- **Improve Code Quality**: Clear interfaces and comprehensive testing
- **Enable Innovation**: Easy to experiment with new components
- **Reduce Risk**: Better validation and error handling
- **Facilitate Growth**: Modular architecture supports complex strategies
- **Support Research**: A/B testing and performance attribution

The component-based architecture is now ready for integration with existing strategies and further development of the trading system.

---

**Related Issues**: Strategy System Redesign Specification  
**Breaking Changes**: None (new components, existing code unchanged)  
**Documentation**: Comprehensive docstrings and test coverage

### CODE REVIEW

**Comment by alexflorisca** [comment]
@codex review

**Comment by alexflorisca** [comment]
@codex review

File: src/strategies/components/strategy_manager.py [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+            # Detect current market regime
+            regime = self.regime_detector.detect_regime(df, index)
+            
+            # Generate trading signal
+            signal = self.signal_generator.generate_signal(df, index, regime)
+            
+            # Calculate risk amount
+            risk_amount = self._calculate_risk_amount(balance, signal, regime)
+            
+            # Calculate position size
```
Comment (alexflorisca) [inline src/strategies/components/strategy_manager.py:150]: ✅ **Fixed**: Now properly integrating RiskManager into strategy execution flow.

**Changes made:**
- `execute_strategy` now calls `self.risk_manager.calculate_position_size()` directly
- The RiskManager's position size is passed to the PositionSizer as the "risk amount" 
- This allows the PositionSizer to further adjust the RiskManager's decision
- Updated metadata to track `risk_position_size` from RiskManager
- Added comprehensive test to verify RiskManager integration

**Architecture Flow:**
1. **RiskManager** calculates initial position size based on signal + regime
2. **PositionSizer** can further adjust that size (using it as "risk amount")
3. **Final validation** ensures reasonable bounds

This ensures that swapping RiskManager implementations actually changes behavior, fulfilling the component-based architecture promise.

**Review by alexflorisca — COMMENTED** [review]
## 🔧 Fixed RiskManager Integration Issue

I've addressed the unresolved comment about the `RiskManager` not being properly wired into the strategy execution flow.

### **Problem Identified**
The review correctly pointed out that:
- `RiskManager.calculate_position_size()` was never called
- Risk calculation was done via private `_calculate_risk_amount()` method
- `RiskManager.should_exit()` and `get_stop_loss()` methods were dead code
- This defeated the purpose of component-based architecture

### **Solution Implemented**

**1. Proper RiskManager Integration:**
```python
# OLD: Bypassed RiskManager
risk_amount = self._calculate_risk_amount(balance, signal, regime)
position_size = self.position_sizer.calculate_size(signal, balance, risk_amount, regime)

# NEW: Uses RiskManager directly
risk_position_size = self.risk_manager.calculate_position_size(signal, balance, regime)
position_size = self.position_sizer.calculate_size(signal, balance, risk_position_size, regime)
```

**2. Component Flow Architecture:**
- **RiskManager** → Calculates initial position size based on signal + regime
- **PositionSizer** → Can further adjust RiskManager's decision (using it as "risk amount")
- **Final Validation** → Ensures reasonable bounds and respects zero positions

**3. Enhanced Testing:**
- Added `test_risk_manager_integration()` to verify RiskManager is actually called
- Validates that different RiskManager implementations will change behavior
- All 183 existing tests still pass

### **Impact**
✅ **Component-based architecture now works as intended**  
✅ **Swapping RiskManager implementations actually changes behavior**  
✅ **Maintains backward compatibility with existing tests**  
✅ **Preserves the previous fix for position sizer zero handling**

The `RiskManager` is now a first-class citizen in the strategy execution pipeline, enabling true component-based risk management as described in the PR.

**Comment by alexflorisca** [comment]
@codex review
